### ISSUE TITLE
Cannot use escape "\" in destination filtering

### ISSUE THREAD

**Issue by Polyhedron-Zero** [SpectoLabs/hoverfly#912]
## Description of the bug
When setting destination filtering using a regular expression, an escape character will result in error, but dry-run is successful.

### Observed result
On Linux shells:
```
# Escape (or double escape) will result in error
$ hoverctl destination "google\.com"
Could not set destination
Malformed JSON
$ hoverctl destination "google\\.com"
Could not set destination
Malformed JSON
$ hoverctl destination "\d+\.com"
Could not set destination
Malformed JSON
$ hoverctl destination "\\d+\\.com"
Could not set destination
Malformed JSON

# However, dry-run works with both
$ hoverctl destination "google\.com" --dry-run=https://www.google.com
The regex provided matches the dry-run URL
$ hoverctl destination "google\\.com" --dry-run=https://www.google.com
The regex provided matches the dry-run URL
$ hoverctl destination "\d+\.com" --dry-run=https://www.163.com
The regex provided matches the dry-run URL
$ hoverctl destination "\\d+\\.com" --dry-run=https://www.163.com
The regex provided matches the dry-run URL
```
On Windows shells, single escape behaves the same, but double escape behaves differently:
```
$ hoverctl destination "google\\.com"
Hoverfly destination has been set to google\.com

$ hoverctl destination "google\\.com" --dry-run=https://www.google.com
The regex provided does not match the dry-run URL
$ hoverctl destination "google\\.com" --dry-run=https://www.google\.com
The regex provided matches the dry-run URL
```

### Workaround
Explicitly use character is working:
```
$ hoverctl destination "[0-9]+[.]com"
Hoverfly destination has been set to [0-9]+[.]com
$ hoverctl destination "[0-9]+[.]com" --dry-run=https://www.163.com
The regex provided matches the dry-run URL
```
But the behavior is still different between set destination and dry-run.

## Additional relevant information
Hoverfly version: 
```
+----------+-------------+
| hoverctl | master-3441 |
| hoverfly | v1.1.4      |
+----------+-------------+
```

**Comment by tommysitu** [comment]
Thank you for the details. It seems like the escape character somehow breaks the JSON body while putting to the destination API. Dry run doesn't have this issue because it's done locally.

**Comment by tommysitu** [comment]
fix released in v1.1.5
